### DISCUSSION TITLE
Problems with Visual Studio

### DISCUSSION THREAD

**Discussion by Zitchas** [endless-sky/endless-sky#10145] [General]
I updated Visual Studio a few hours ago, and now I can't build ES any more.

Visual Studio gives the following message:

`CMake Error: CMake was unable to find a build program corresponding to "Ninja Multi-Config".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.`

Is anyone else having this problem? Or know of how to fix it?

**Comment by warp-core** [comment]
- https://github.com/microsoft/vcpkg/issues/39048

This would suggest deleting the vcpkg folder inside your endless-sky directory should fix the issue.
That doesn't seem to work for me with VS 17.10.1, though.
If that doesn't work for you, I suggest rolling back Visual Studio for now. You can do this through the Visual Studio Installer with the "more" menu below launch and modify. 17.8.3 is the previous version I had which works for me, and the linked issue suggests that 17.9.x also worked.

Edit: after a little bit more reading, I think there might be some sort of conflict between Visual Studio 2022 Community version numbers and hte VS build tool version numbers, which leads vcpkg to discard the Visual Studio 2022 Community path and then it can't find the Visual Studio installation. This is after skimming one sentence from an report from almost a year ago, though, so maybe not. If this is the issue, downgrading or completely removing those build tools or just waiting for another update to Visual Studio itself (or installing the 17.11 preview) might work. I haven't tried any of that, though.

**Reply by Zitchas** [reply]
I opted to fully delete the entire Endless Sky folder and then recreate the local ES git directory; which did not result in any change in the problem. 

I was on 17.9.x, so that was working for me.

Thanks for locating that.

**Reply by Zitchas** [reply]
Tried the rollback, and that just generated more (different) problems, so it appears that functionality doesn't fully rollback everything; or it leaves stuff in the preferences that result in erroneous expectations. 

I might try the preview build at some point.
